File: finance-api/src/main/java/com/financesystem/finance_api/modules/tenant/fx/domain/model/OperationFee.java
package com.financesystem.finance_api.modules.tenant.fx.domain.model;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record OperationFee(
        UUID id,
        FxOperationCode operationCode,
        FeeType feeType,
        BigDecimal feeValue,
        FeeCalculationMode calculationMode,
        boolean active,
        String description,
        Instant createdAt,
        Instant updatedAt
) {
}

File: finance_mobile/lib/domain/entities/operation_fee.dart
class OperationFee {
  final String id;
  final String operationCode;
  final String feeType;
  final double feeValue;
  final String calculationMode;
  final bool active;
  final String? description;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const OperationFee({
    required this.id,
    required this.operationCode,
    required this.feeType,
    required this.feeValue,
    required this.calculationMode,
    required this.active,
    required this.description,
    required this.createdAt,
    required this.updatedAt,
  });
}

class OperationFeesPage {
  final List<OperationFee> items;
  final int totalPages;
  final int totalElements;
  final int number;
  final int size;
  final bool first;
  final bool last;

  const OperationFeesPage({
    required this.items,
    required this.totalPages,
    required this.totalElements,
    required this.number,
    required this.size,
    required this.first,
    required this.last,
  });
}
